packages/xo-web/src/common/xo/migrate-vm-modal/index.js

```diff
@@ -236,79 +406,113 @@ export default class MigrateVmModalBody extends BaseComponent {
-                  <Col size={6}>
-                    <SelectNetwork
-                      onChange={this._selectMigrationNetwork}
-                      predicate={this._getMigrationNetworkPredicate()}
-                      value={migrationNetworkId}
-                    />
+          </div>
+        )}
+        {!isEmpty(snapshots) &&
+          (displayVdis || displayVifs) && (
+            <div className={styles.groupBlock}>
+              <div>
+                <Row>
+                  <Col>
+                    <strong> {_('IndividualSnapshotConfiguration')}:</strong>
                   </Col>
-                </SingleLineRow>
+                </Row>
               </div>
-              <div className={styles.groupBlock}>
-                <SingleLineRow>
-                  <Col>{_('migrateVmSelectNetworks')}</Col>
-                </SingleLineRow>
-                <br />
-                <SingleLineRow>
-                  <Col size={6}>
-                    <span className={styles.listTitle}>
-                      {_('migrateVmVif')}
-                    </span>
-                  </Col>
-                  <Col size={6}>
-                    <span className={styles.listTitle}>
-                      {_('migrateVmNetwork')}
-                    </span>
-                  </Col>
-                </SingleLineRow>
-                {map(vifs, vif => (
-                  <div className={styles.listItem} key={vif.id}>
-                    <SingleLineRow>
-                      <Col size={6}>
-                        {vif.MAC} ({networks[vif.$network].name_label})
-                      </Col>
-                      <Col size={6}>
-                        <SelectNetwork
-                          onChange={network =>
-                            this.setState({
-                              mapVifsNetworks: {
-                                ...mapVifsNetworks,
-                                [vif.id]: network.id,
-                              },
-                            })
-                          }
-                          predicate={this._getTargetNetworkPredicate()}
-                          value={mapVifsNetworks[vif.id]}
-                        />
-                      </Col>
-                    </SingleLineRow>
-                  </div>
+              <div>
+                {map(snapshots, snapshot => (
+                  <Collapse
+                    buttonText={snapshot.name_label}
+                    size='small'
+                    className='mb-1'
+                  >
+                    <Snapshot
+                      displayVdis={displayVdis}
+                      displayVifs={displayVifs}
+                      key={snapshot.id}
+                      mapVdisSrs={mapVdisSrs}
+                      mapVifsNetworks={mapVifsNetworks}
+                      networkPredicate={this._getTargetNetworkPredicate()}
+                      onChange={props => {
```

Comment:
This may be a good idea to pass the `linkState` to `Snapshot` so you don't have to handle the painful and dangerous merges.